src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesSynchronisationService.kt

```diff
@@ -213,22 +213,19 @@ Also add new mappings for the new booking id for the copied case notes, which po
      */
     val (nomsNumber, removedNomsNumber, bookingId) = prisonerMergeEvent.additionalInformation
     try {
-      val nomisCaseNotes = nomisApiService.getCaseNotesForPrisoner(nomsNumber)
-        .caseNotes
+      val nomisCaseNotes = nomisApiService.getCaseNotesForPrisoner(nomsNumber).caseNotes
       val freshlyMergedNomisCaseNotes = nomisCaseNotes
-        .filter {
-          isMergeCaseNoteRecentlyCreated(it, prisonerMergeEvent)
-        }
+        .filter { isMergeCaseNoteRecentlyCreated(it, prisonerMergeEvent) }
         .apply {
           if (isEmpty()) {
             throw IllegalStateException("Merge data not ready for $nomsNumber")
           }
         }
 
       val existingMappings = caseNotesMappingService.getMappings(nomisCaseNotes.map { it.caseNoteId })
+        .associateBy { it.nomisCaseNoteId }
```

Comment:
now create a map (keyed by the unique `nomisCaseNoteId` of all the existing mappings